Fixed #30913 -- Added support for covering indexes on PostgreSQL 11+. (#11991)
django/django: docs/ref/models/constraints.txt

These conditions have the same database restrictions as
:attr:`Index.condition`.

``include``

> -------------

Reviewer: Please fix the heading underlines to be the same length as the heading itself.

.. attribute:: UniqueConstraint.include

.. versionadded:: 3.1

> A list of fields to be included with the unique index. This allows for fully

Reviewer: `...included with...` → `...include as non-key columns on...`

covered indexes to be used by filtering on the unique field and selecting

> the included fields with :meth:`~django.db.models.query.QuerySet.only()`.

Reviewer: Perhaps we should have an example of using `QuerySet.only()`?

    Deferred unique constraints may lead to a `performance penalty
    <https://www.postgresql.org/docs/current/sql-createtable.html#id-1.9.3.85.9.4>`_.

> ``include``

Reviewer 2: Is include supported just on Postgres? Is that said anywhere, other than 3.2.txt?

Reviewer 3: I added note:
```
``include`` is supported only on PostgreSQL.
```

-----------

.. attribute:: UniqueConstraint.include

.. versionadded:: 3.2

> A list or tuple of the name of the fields to be included in the covering unique

Reviewer 2: ```suggestion
A list or tuple of the names of the fields to be included in the covering unique
```

index as non-key columns. This allows index-only scans to be used for queries
that select only included fields (:attr:`~UniqueConstraint.include`) and filter
only by unique fields (:attr:`~UniqueConstraint.fields`).

For example::

    UniqueConstraint(name='unique_booking', fields=['room', 'date'], include=['full_name'])

> will allow filtering on ``room`` and ``date``, and also selecting ``full_name``
> with having to only fetch data from the index.

Reviewer 2: ```suggestion
will allow filtering on ``room`` and ``date``, also selecting ``full_name``,
while fetching data only from the index.
```
